catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/reader/GetRecordsMessageBodyReader.java

```diff
@@ -92,6 +97,24 @@ public CswRecordCollection readFrom(Class<CswRecordCollection> type, Type generi
             MultivaluedMap<String, String> httpHeaders, InputStream inStream)
             throws IOException, WebApplicationException {
 
+        CswRecordCollection cswRecords = null;
+
+        // If the following HTTP header exists and its value is true, the input stream will contain
+        // raw product data
+        List<String> productRetrievalHeader =
+                httpHeaders.get(CswConstants.PRODUCT_RETRIEVAL_HTTP_HEADER);
+        if (productRetrievalHeader != null) {
+            if (productRetrievalHeader.get(0)
+                    .equals("true")) {
+                inStream = new ByteArrayInputStream(IOUtils.toByteArray(inStream));
+                String fileName = handleContentDispositionHeader(httpHeaders.
+                        get(HttpHeaders.CONTENT_DISPOSITION)
+                        .get(0));
+                cswRecords = new CswRecordCollection();
+                cswRecords.setResource(new ResourceImpl(inStream, mediaType.toString(), fileName));
```

Comment:
Is there a reason you couldn't use the original, unshadowed `inStream` here? Is it to prevent the input stream from escaping here or is it something else?